src/nci_util.c

```diff
@@ -912,6 +1040,173 @@ nci_discovery_ntf_copy(
     }
 }
 
+/*==========================================================================*
+ * Interface
+ *==========================================================================*/
+
+NciModeParam*
+nci_mode_param_copy(
+    const NciModeParam* param,
+    NCI_MODE mode)
+{
+    if (G_LIKELY(param)) {
+        NciModeParam* copy = NULL;
+        gsize size = nci_mode_param_size(param, mode);
+
+        switch (mode) {
+        case NCI_MODE_ACTIVE_POLL_A:        /* fallthrough */
+        case NCI_MODE_PASSIVE_POLL_A:       /* fallthrough */
+        case NCI_MODE_ACTIVE_POLL_F:        /* fallthrough */
+        case NCI_MODE_PASSIVE_POLL_F:       /* fallthrough */
+        case NCI_MODE_PASSIVE_POLL_B:       /* fallthrough */
+        case NCI_MODE_ACTIVE_LISTEN_F:      /* fallthrough */
+        case NCI_MODE_PASSIVE_LISTEN_F:     /* fallthrough */
```

Comment:
ETOOMANYFALLTHROUGHS. Isn't that obvious that it's an intentional fallthrough? To me they just look disturbing and don't add anything to readability. Or even do the opposite.